fixed issues on java troubleshooting page (#990)
couchbaselabs/docs-couchbase-lite: modules/java/pages/troubleshooting-logs.adoc

It is enabled by default.
To change default settings use database's https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/Log.html[Log] method to set the required values -- see xref:java:troubleshooting-logs.adoc#eg-cons-log[]

> You'll primarily use `Database.log.getConsole()` to get the https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html[ConsoleLogger] to control console logging.

Reviewer: Did you want to remove the first link on this?

Author: Yes, we do not use the `log.getConsole()` to control console logging.

====

> <.> Set the domains to be logged. 

Reviewer: ```suggestion
<.> Set the domains you want to log. 
```

Related::
https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/Log.html[Log] | https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/Log.html[log.getConsole()] | https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html[ConsoleLogger]

[#lbl-file-logs]
== File based logging
File based logging defaults to off --  see: xref:java:troubleshooting-logs.adoc#eg-file-log[] for how to enable it.

Use https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/Log.html[log.getFile()] and https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/FileLogger.html[FileLogger] to control file-based logging.

[#formats]
=== Formats
Available file based logging formats:

* Binary -- most efficient for storage and performance.

> Binary format defaults for file based logging.

Reviewer: ```suggestion
Binary format is the default format for file based logging.
```

====

> <.> Set the domains to be logged. Here we turn on logging for all available domains -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html#setDomains-java.util.EnumSet-[log.getConsole().setDomain()] and enum https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/LogDomain.html[LogDomain]

Reviewer: ```suggestion
<.> Set the domains you want to log. The following example turns on logging for all available domains -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html#setDomains-java.util.EnumSet-[log.getConsole().setDomain()] and enum https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/LogDomain.html[LogDomain]
```

Reviewer: Realizing looking at this again now that you're using the code annotations. Style doesn't want to use these, so maybe thinking about cleaning them up later? 
```suggestion
<.> Set the domains you want to log. This example turns on logging for all available domains -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html#setDomains-java.util.EnumSet-[log.getConsole().setDomain()] and enum https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/LogDomain.html[LogDomain]
```

> <.> Set the domains to be logged. The following example turns on logging for all available domains -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html#setDomains-java.util.EnumSet-[log.getConsole().setDomain()] and enum https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/LogDomain.html[LogDomain]

Reviewer: ```suggestion
<.> Set the domains you want to log. This example turns on logging for all available domains -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/ConsoleLogger.html#setDomains-java.util.EnumSet-[log.getConsole().setDomain()] and enum https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/LogDomain.html[LogDomain]
```

Reviewer: The first part was the passive voice. "to be logged" - fix that up and we're golden.

Author: Done.

====

<.> Set the log file directory
<.> Change the max rotation count from the default (1) to 5. +
*Note* this means six files may exist at any one time; the five rotated log files, plus the active log file
<.> Set the maximum size (bytes) for the log file

> <.> Select the binary log format (included for reference only as binary format defaults)

Reviewer: ```suggestion
<.> Select the binary log format (included for reference only as binary format is the default)
```

> <.> Increase the log output level from the default (_warnings_) to _info_ -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/FileLogger.html#setLevel-com.couchbase.lite.LogLevel-[log.getFile().setLevel()]

Reviewer: ```suggestion
<.> Increase the log output level from the default (`warnings`) to `info` -- see: https://docs.couchbase.com/mobile/{major}.{minor}.{maintenance-java}{empty}/couchbase-lite-java/com/couchbase/lite/FileLogger.html#setLevel-com.couchbase.lite.LogLevel-[log.getFile().setLevel()]
```
